core/controllers/reader.py

```diff
@@ -903,10 +904,55 @@ class RecommendationsHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    @acl_decorators.can_access_story_viewer_page
+    def _record_completion_and_get_next_exp_id(
+            self, story_id, current_node_id):
+        """This part is split into a separate function so as to reuse the
+        'can_access_story_viewer_page' decorator to verify whether the user is
+        indeed playing an exploration in a published story and topic.
+        This function records the chapter corresponding to current_node_id as
+        completed and returns the exploration ID of the next chapter in the
+        story.
+
+        Args:
+            story_id: str. The story ID in which the chapter exists.
+            current_node_id: str. The node ID of the chapter that the user just
+                completed.
+
+        Returns:
+            str. The exploration ID of the next chapter in the story.
```

Comment:
How are you treating the "last chapter in the story" case? This is not documented, and should be.

The docstring is also missing a Raises section.

Reply:
Done and added a return None for the last chapter case.